drv/ArrayList.drv

```diff
@@ -571,6 +571,20 @@ public class ARRAY_LIST KEY_GENERIC extends ABSTRACT_LIST KEY_GENERIC implements
 		return modified;
 	}
 
+	@Override
+	public boolean retainAll(final COLLECTION c) {
+		final KEY_TYPE[] a = this.a;
+		int j = 0;
+		for(int i = 0; i < size; i++)
+			if (c.contains(a[i])) a[j++] = a[i];
+#if KEYS_REFERENCE
+		Arrays.fill(a, j, size, null);
+#endif
+		final boolean modified = size != j;
```

Comment:
@vigna More general question: Maybe the list should trim() here if it significantly decreased in size?